### DISCUSSION TITLE
Some question about train and validation and inference

### DISCUSSION THREAD

**Discussion by okt7760** [fcakyon/small-object-detection-benchmark#18] [Q&A] [answered]
Hi @fcakyon ,I have some trouble in training about the detail of Training details,so I was hoping you could offer some help in achieving the results you've announced
The first question concerns the processing of visdrone datasets,the val dataset is need to be slide,slice_visdrone.py use in val or training or testing?
Second question,i am use your checkpoints from :
https://github.com/fcakyon/sahi-benchmark/releases/download/v0.0.2/tood_sf_visdrone.pth (TOOD+SF+SAHI)
and config use :
https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_configs/visdrone_tood/tood_crop_480_960_cls_60.py
use the code 
https://github.com/fcakyon/small-object-detection-benchmark/blob/main/eval_tools/predict_evaluate_analyse.py
we have very poor result 
We find the console constantly display ignore unverified…………，so we consider the problem of visdrone, but we use your tool visdrone_to_coco.py and no changes were made
hope you can help me these two problems

**Comment by fcakyon** [comment, marked as answer]
> The first question concerns the processing of visdrone datasets,the val dataset is need to be slide,slice_visdrone.py use in val or training or testing?

You have to only convert the visdrone  training and validation splits into coco format using [visdrone_to_coco](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/visdrone/visdrone_to_coco.py). No need to slice the dataset if you use the config files from this repo.

**Reply by okt7760** [reply]
Thank you for your reply!

The Validation splits do not require slice with [slice_visdrone](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/visdrone/slice_visdrone.py) scirpt ?  

I used the slice visdrone validation split in the training process,Is this a wrong step? 

For validation split operations, only coco format is required, and slice is not required?

**Reply by fcakyon** [reply]
> The Validation splits do not require slice with [slice_visdrone](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/visdrone/slice_visdrone.py) script ?

Yes, it does not require.

> I used the slice visdrone validation split in the training process,Is this a wrong step?

Yes, it is wrong.

> For validation split operations, only coco format is required, and slice is not required?

Yes, it is true.

**Comment by fcakyon** [comment]
> Second question,i am use your checkpoints from : https://github.com/fcakyon/sahi-benchmark/releases/download/v0.0.2/tood_sf_visdrone.pth (TOOD+SF+SAHI) and config use : https://github.com/fcakyon/small-object-detection-benchmark/blob/main/mmdet_configs/visdrone_tood/tood_crop_480_960_cls_60.py use the code https://github.com/fcakyon/small-object-detection-benchmark/blob/main/eval_tools/predict_evaluate_analyse.py we have very poor result We find the console constantly display ignore unverified…………，so we consider the problem of visdrone, but we use your tool visdrone_to_coco.py and no changes were made

You are not slicing the visdrone validation split, right? 

You convert the validation split with [visdrone_to_coco](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/visdrone/visdrone_to_coco.py) script, right? 

What is your `sahi` version?

**Reply by okt7760** [reply]
Thanks your reply!
sahi version : 0.11.0
First i am use the [slice_visdrone](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/visdrone/slice_visdrone.py) script to slice validation file (VisDrone2019-DET-val) after that convert the validation with [visdrone_to_coco](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/visdrone/visdrone_to_coco.py) script,Is this the right step ?

**Reply by fcakyon** [reply]
It is wrong. You have to convert the validation split without slicing it. No manual slicing is required.

**Reply by okt7760** [reply]
According to your reply, I have revised it and it is close to the result on the paper. Thanks for your help to solve my problem :blush: !!

**Reply by fcakyon** [reply]
I am glad to hear that!

**Reply by bshakya77** [reply]
Hello @fcakyon nad @okt7760,

I am trying to recreate the same process to predict and analyse (TOOD + SF) model on VisDrone2019 dataset. The process I followed was:

```
Config:

Pytorch: 1.10.0
CUDA: 11.3
Python: 3.9.21
mmdet: 3.1.0
mmcv: 2.0.0
```

1. Downloaded the (TOOD + SF) model:` tood_sf_visdrone.pth.`
2. Converted the VisDrone2019 dataset to COCO format using `visdrone_to_coco.py` file. I only generated json files for train, val, and test dataset and used the same image folder.
3. Use [predict_evaluate_analyse.py](https://github.com/fcakyon/small-object-detection-benchmark/blob/main/eval_tools/predict_evaluate_analyse.py) to perfrom the prediction and analysis.
4. I used the following config file for (TOOD + SF) model. I converted the config file to version 3.x because when I used mmdet=2.1.0 version, it returned mmdet >= 3.0.0 error.
5. I got the following error when I ran : `predict_evaluate_analyse.py`

![image](https://github.com/user-attachments/assets/3d1f96c2-91da-405b-aef2-50e7351bae6d)

I have attached all the required files in zip folder. Could you please review if error is because of config file or dataset file I used.

Also, could you provide me with the required config file for (TOOD + SF) model on VisDrone dataset for mmdet version 3.x?

I highly appreciate your help.

Regards,
Bijay

**Reply by fcakyon** [reply]
@bshakya77 unfortunately this repo doesnt work with mmdet 3.0.0. check here for requirements: https://github.com/fcakyon/small-object-detection-benchmark/blob/main/requirements.txt

**Reply by bshakya77** [reply]
Hello @fcakyon,

Thank you, it worked after installing the provided requirements.

I am able to produce the evaluation results for FCOS and TOOD model as provided in the repository. Could you please provide the VFNET model trained on VisDrone2019 dataset since it is not available in this repository.

Also, I was trying to compare the inference time for various models with SAHI. So, which inference time is the optimal inference time to benchmark the results.

I appreciate your feedback.

Regards,
Bijay
